src/Tools/Utils.php

```diff
@@ -190,13 +192,31 @@ public static function getModelFactory(string $modelName, array $states = [], ar
             /** @var \Illuminate\Database\Eloquent\Factories\Factory $factory */
             $factory = call_user_func_array([$modelName, 'factory'], []);
             foreach ($states as $state) {
-                $factory = $factory->$state();
+                if (method_exists(get_class($factory), $state)) {
```

Comment:
Why the wrapper here? What if the user made a typo in the state? Won't we be hiding errors?

Reply:
Ah, I see. Because we're passing the states along in the recursive call, so they may/may not apply to the entire relation chain. Not a perfect solution, but it'll pass.